src/iter/mod.rs

```diff
@@ -2339,6 +2340,41 @@ pub trait ParallelIterator: Sized + Send {
         SkipAnyWhile::new(self, predicate)
     }
 
+    /// Collects this iterator into a linked list of vectors.
+    ///
+    /// This is useful when you need to condense a parallel iterator into a collection,
+    /// but have no specific requirements for what that collection should be. If you
+    /// plan to store the collection longer-term, `Vec<T>` is, as always, likely the
+    /// best default choice, despite the overhead that comes from concatenating each
+    /// vector. Or, if this is an `IndexedParallelIterator`, you should also prefer to
+    /// just collect to a `Vec<T>`.
+    ///
+    /// Internally, most [`FromParallelIterator`]/[`ParallelExtend`] implementations
+    /// use this strategy; each job collecting their chunk of the iterator to a `Vec<T>`
+    /// and those chunks getting merged into a `LinkedList`, before then extending the
+    /// collection with each vector. This is the most efficient way to collect an
+    /// unindexed parallel iterator (again, indexed parallel iterators can be
+    /// efficiently collected simply into a vector).
```

Comment:
I'm hesitant about the superlative, and I think we don't need the repeated statement about indexed iterators. How about this, adding a clue about *why* it's efficient:
```suggestion
    /// collection with each vector. This is a very efficient way to collect an
    /// unindexed parallel iterator, without much intermediate data movement.
```

Reply:
That makes sense - the superlative felt wrong, but I wasn't sure how to word the why of it.